### PR TITLE
Support ELF Tool Chain and LLVM objdump

### PR DESCRIPTION
**Pull request by jrtc27**

### CODE REVIEW

**Comment by partouf** [comment]
I have no comments, this looks excellent.

(Not looking forward to testing this on all languages though /cry)

**Review by partouf — APPROVED** [review]

**Comment by mattgodbolt** [comment]
Wow, this is great! Thanks @jrtc27 ! Taking a look now :)

File: lib/compilers/assembly.js [hunk 1]
```diff
@@ -87,20 +87,7 @@ export class AssemblyCompiler extends BaseCompiler {
         return this.postProcess(asmResult, outputFilename, filters);
     }
 
-    async objdump(outputFilename, result, maxSize, intelAsm, demangle) {
```
Comment (mattgodbolt) [inline lib/compilers/assembly.js:90]: Yay for removing all this code! Thank you

File: lib/objdumper/index.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+// POSSIBILITY OF SUCH DAMAGE.
+
+import { makeKeyedTypeGetter } from '../keyed-type';
+
+import * as all from './_all';
+
+export { BaseObjdumper } from './base';
+export * from './_all';
+
+export const getObjdumperTypeByKey = makeKeyedTypeGetter('objdumper', all);
```
Comment (mattgodbolt) [inline lib/objdumper/index.js:32]: Ah @apmorton like you said on the Discord, someone else using your `makeKeyedTypeGetter`! :)

File: test/base-compiler-tests.js [hunk 3]
```diff
@@ -492,6 +493,34 @@ describe('Compiler execution', function () {
             true,
             true);
         result.asm.should.deep.equal('the output');
+    }
+
+    it('should run default objdump properly', async () => {
```
Comment (mattgodbolt) [inline test/base-compiler-tests.js:498]: Love it, thanks.

**Review by mattgodbolt — APPROVED** [review]
Wow, this is amazing. I usually find some pedantic thing to say about PRs this big, but not this time! Thank you so much! Like @partouf says, we'll need to do a bit of testing in our staging environment to ensure it works.

Are there any suggestions you have for changing any of the `compiler-explorer.amazon.properties` to use anything other than the default objdumper? Is this PR mainly for local instances to be able to override and use llvm's dumper? It seems we could usefully expose llvm's objdump as an option at some later date, pending some UI work on our side.

**Comment by mattgodbolt** [comment]
Well, thanks! We appreciate it very much. I'll merge later when I have a little spare time to push it through to our staging environment and check it out a little. This PR was a joy to review: thank you!

**Comment by mattgodbolt** [comment]
By the way, I recommend you ping us off github about running a public-accessible CE instance. We have some war stories and thoughts on securing it as much as possible. As you are no doubt aware, a compiler is amazing attack vector to make accessible to the world :) (though a cursory examination seems like you've done sensible things).

**Comment by mattgodbolt** [comment]
(and by off-GH I mean feel free to email me matt at the obvious domain, or join us on Discord: https://discord.gg/gagKdZF5

**Comment by mattgodbolt** [comment]
Merging and testing now! Thanks and sorry for the slowness here..

**Pull request merged**